packages/@expo/cli/src/utils/analytics/rudderstackClient.ts

```diff
@@ -96,13 +98,20 @@ export async function logEventAsync(
     return;
   }
 
-  await getUserAsync().catch(() => null);
-  ensureIdentified();
+  // this has the side effect of calling `setUserData` which fetches the user and populates identifyData
+  try {
+    await getUserAsync();
```

Comment:
I think in the case where the user is not authenticated, we'd skip setUserData and still log an event, but without the anonymous id set to anything?

Reply:
no id causes crashes on rudder server. we'd need to generate a random id or drop the event